File: python/spike.py

import math
import os
import time

DELAY = 0.1

def main():
    while True:
        for i in range(1, int(math.sqrt(os.get_terminal_size()[0] - 1))):
            print('-' * (i*i))
            time.sleep(DELAY)
        for i in range(int(math.sqrt(os.get_terminal_size()[0] - 1)), 1, -1):
            print('-' * (i*i))
            time.sleep(DELAY)


try:
    main()
except KeyboardInterrupt:
    print('Spike, by Al Sweigart al@inventwithpython.com')

File: typescript/spike.ts
// To get ts types, run: npm i --save-dev @types/readline

namespace Spike {
    let width: number = 120;
    const DELAY: number = 100;

    async function sleep(milliseconds: number): Promise<void> {
        return new Promise(resolve => setTimeout(resolve, milliseconds));
    }

    process.on('SIGINT', function() {
        console.log("\nSpike by Al Sweigart al@inventwithpython.com");
        process.exit();
    });

    function getTerminalWidth(): number {
        if (process.stdout.isTTY) {
            return process.stdout.columns - 1; // -1 is because Windows puts a newline when printing to the last column
        } else {
            return 80;
        }
    }

    async function main(): Promise<void> {
        while (true) {
            for (let i = 1; i * i < width; i++) {
                width = getTerminalWidth();
                console.log('-'.repeat(i * i));
                await sleep(DELAY);
            }

            for (let i = Math.floor(Math.sqrt(width)) - 1; i > 1; i--) {
                console.log('-'.repeat(i * i));
                await sleep(DELAY);
            }
        }
    }

    main();
}